std/mem.zig

```diff
@@ -72,6 +74,7 @@ pub const Allocator = struct {
         const byte_count = math.mul(usize, @sizeOf(T), n) catch return Error.OutOfMemory;
         const byte_slice = try self.allocFn(self, byte_count, alignment);
         assert(byte_slice.len == byte_count);
+        _ = std.valgrind.memcheck.makeMemDefined(byte_slice);
```

Comment:
This is not correct - the memory is undefined here, as you can see from the for loop directly below. I'm adding support for this in the language directly, so this call is unnecessary.

Reply:
> This is not correct - the memory is undefined here, as you can see from the for loop directly below.

We have to set it as defined here or the loop directly below ends up throwing valgrind errors!

> I'm adding support for this in the language directly, so this call is unnecessary.

The support in the language works for e.g. marking variables on the stack as undefined. However I don't see how it can cover custom "userspace" allocators?

Reply:
You want `VALGRIND_MAKE_MEM_UNDEFINED`:

> /* Similarly, mark memory at _qzz_addr as addressable but undefined
   for _qzz_len bytes. */

...which is exactly what the loop is doing. (Side note- we can potentially have `@memset` recognize when the value to set is undefined and make this more efficient)